packages/analytics-js-plugins/src/externalAnonymousId/util.js

```diff
@@ -0,0 +1,24 @@
+import {
+  defaultCookieStorage,
+  defaultLocalStorage,
+} from '@rudderstack/analytics-js/services/StoreManager/storages';
+import { externallyLoadedSessionStorageKeys } from '@rudderstack/analytics-js/components/userSessionManager/sessionStorageKeys';
+
+const getSegmentAnonymousId = () => {
+  let anonymousId;
+  /**
+   * First check the local storage for anonymousId
+   * Ref: https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#identify
+   */
+  if (defaultLocalStorage.isSupportAvailable) {
```

Comment:
I think we can reuse the getStorageEngine from StoreManager L67

Reply:
Done